File: MPPCS/Service/ShowService.cs
using CommonDomain;
using Persistence.Repository;
using Persistence.Service.Validator;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Persistence.Service
{
    public class ShowService
    {
        private ShowRepository repository;
        private ShowValidator validator;

        public ShowService(ShowRepository repository)
        {
            this.repository = repository;
            this.validator = new ShowValidator();
        }

        public List<Show> GetAllShows()
        {
            return repository.GetAll();
        }

        public List<Show> GetShowsByDate(DateTime time)
        {
            List<Show> showsOnDate = (List<Show>)repository.GetAll().Where
                (show => show.GetStartTime().Year == time.Year &&
                         show.GetStartTime().Month == time.Month &&
                         show.GetStartTime().Day == time.Day
                ).ToList<Show>();

            return showsOnDate;
        }

        public void AddShow(Show show)
        {
            validator.validate(show);
            repository.Add(show);
        }

        public Show FindShow(int showId)
        {
            return repository.Find(showId);
        }

        public Show UpdateShow(Show newShow)
        {
            validator.validate(newShow);
            return repository.Update(newShow);
        }
    }
}

File: PersistenceAndServices/src/main/java/service_base/ShowService.java
package service_base;

import domain.Show;
import repository.ShowDatabaseRepository;
import service_base.validators.ShowValidator;

import java.time.LocalDate;
import java.util.List;
import java.util.stream.Collectors;

public class ShowService {
    private ShowDatabaseRepository repository;
    private final ShowValidator validator;

    public ShowService(ShowDatabaseRepository repository) {
        this.repository = repository;
        this.validator = new ShowValidator();
    }

    public List<Show> getAllShows() throws Exception {
        return repository.getAll();
    }

    public List<Show> getShowsByDate(LocalDate date) throws Exception {

        List<Show> showsOnDate = repository.getAll()
                .stream()
                .filter(show -> show.getStartTime().getYear()==date.getYear()&&
                                show.getStartTime().getMonthValue()==date.getMonthValue()&&
                                show.getStartTime().getDayOfMonth()==date.getDayOfMonth())
                .collect(Collectors.toList());

        return showsOnDate;
    }

    public void addShow (Show show) throws Exception {
        validator.validate(show);
        repository.add(show);
    }

    public Show findShow(int showId) throws Exception {
        return repository.find(showId);
    }

    public Show updateShow(Show newShow) throws Exception {
        validator.validate(newShow);
        return repository.update(newShow);
    }
}
